src/components/containers/Disclosure/Disclosure.module.scss

```diff
@@ -0,0 +1,140 @@
+    background: var(--bk-disclosure-background-color);
+    color: var(--bk-disclosure-text-color);
+    
+    // Use flex so that we can fit the content to the <details> height, allowing us to do `overflow-y: auto`
+    display: flex;
+    flex-direction: column;
+    
+    summary {
+      @include heading.h5;
+      
+      cursor: pointer;
+      
+      // Expand the target size
+      margin: calc(-1 * var(--bk-disclosure-padding-block)) calc(-1 * var(--bk-disclosure-padding-inline));
+      padding: var(--bk-disclosure-padding-block) var(--bk-disclosure-padding-inline);
+      
+      display: flex;
+      align-items: center;
+      gap: bk.$spacing-9;
+      
+      line-height: bk.$line-height-s;
+      
+      .bk-disclosure__title {
+        @include bk.text-one-line;
+      }
+      
+      .bk-disclosure__collapse-icon {
+        margin-inline-start: auto;
+        font-size: 1.125em; // ~18px
+        
+        @media (prefers-reduced-motion: no-preference) {
+          transition: rotate 120ms ease-in-out;
+        }
+      }
+      
+      // Border radius for focus outline
+      border-radius: // Note: `inherit` keyword does not work for some reason?
+        var(--bk-disclosure-border-radius-top)
+        var(--bk-disclosure-border-radius-top)
+        var(--bk-disclosure-border-radius-bottom)
+        var(--bk-disclosure-border-radius-bottom);
+      @include bk.focus-inset;
+    }
+    &[open] summary {
+      --bk-disclosure-border-radius-bottom: 0; // Disable bottom radius on the summary when open (for focus outline)
+      
+      .bk-disclosure__collapse-icon {
+        rotate: -0.5turn;
+      }
+    }
+    
+    // Inner wrapper for content. We cannot give `::details-content` any margin/padding because it would still be
+    // visible (due to `content-visibility: hidden` still taking up space). So we use this as scroller and for spacing.
+    .bk-disclosure__content {
+      overflow-y: scroll;
+      
+      margin-top: bk.$spacing-8;
+      margin-bottom: calc(bk.$spacing-8 - var(--bk-disclosure-padding-block)); // Total should equal `bk.$spacing-8`
+    }
+    
+    // Animate the collapsing of the details content.
+    // Note: `::details-content` does not yet have full browser support, but it is a progressive enhancement.
+    &::details-content {
+      overflow: hidden;
+      block-size: 0;
+      
+      display: flex;
+      flex-direction: column;
+      
+      @media (prefers-reduced-motion: no-preference) {
+        // Note: do not use `transition` shorthand due to https://github.com/parcel-bundler/lightningcss/issues/572
+        transition-property: block-size, content-visibility;
+        transition-duration: var(--bk-disclosure-transition-duration);
+        transition-behavior: allow-discrete;
+      }
+    }
+    &[open]::details-content {
+      block-size: auto;
+    }
+    
+    // Fallback styling for browsers that don't support `::details-content`
```

Comment:
We don't support transition movement for browsers that don't support `::details-content`?
In firefox, the accordion open/close movement is not smooth like chrome.

In below video, the first test is firefox, the second test is chrome.

https://github.com/user-attachments/assets/6ec1bd52-7d1c-4832-a450-4723cacb3f85

Reply:
Right, I implemented this component using `<details>` rather than inventing my own using `<div>`s and a lot of `aria-` attributes. `<details>` is pretty much perfect for this use case, and has decent accessibility out of the box. The downside though is that Firefox and Safari don't yet support `::details-content` and that means we cannot implement certain things like animating the open/close transition. Safari just landed `::details-content` though so that will be in an upcoming release.

There's a good video on the topic here from Kevin Powell: https://www.youtube.com/watch?v=Vzj3jSUbMtI

My idea here is to make this a progressive enhancement: browsers that support it will get the animation, browsers that don't still get all the working functionality just without the animation.